corehq/apps/app_manager/models.py

```diff
@@ -792,6 +792,43 @@ def short_comment(self):
         return self.comment if len(self.comment) <= 500 else self.comment[:497] + '...'
 
 
+class CaseLoadReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    path = StringProperty()
+    properties = ListProperty(unicode)
+
+
+class CaseSaveReference(DocumentSchema):
+    _allow_dynamic_properties = False
+    case_type = StringProperty()
+    path = StringProperty()
+    properties = ListProperty(unicode)
+    create = BooleanProperty(default=False)
+    close = BooleanProperty(default=False)
+
+
+class CaseReferences(DocumentSchema):
+    _allow_dynamic_properties = False
+    load = DictProperty()
+    save = DictProperty()
```

Comment:
updated in https://github.com/dimagi/commcare-hq/pull/14710/commits/4b5657f21849bf6a31fdeaa04ddf377c019539df and added some doc strings to better explain the intent of these different classes/methods